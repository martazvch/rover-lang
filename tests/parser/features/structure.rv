struct Foo {
    fn bar(self, a: int) {
        print a
    }

    fn buz(c: bool) -> Self {}
}

struct Bar {
    name: int,
    age: str,
    life: float,
    parent: Human,

    number = 1234,
    money: int = 120
}

struct Baz {
    name: int,
    parent: Human,

    number = 1234,
    money: int = 120

    fn bar(self, a: int) {
        print a
    }

    fn buz(c: bool) -> Self {}
}
