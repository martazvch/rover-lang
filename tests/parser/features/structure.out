[Structure declaration Foo
    [Fn declaration bar, type void, arity 2
        params:
            self
            a, type int
        body:
            [Block]
                [Print]
                    [Identifier a]
    ]
    [Fn declaration buz, type Self, arity 1
        params:
            c, type bool
        body:
            [Block]
    ]
]

[Structure declaration Bar
    fields:
        name, type: int
        age, type: str
        life, type: float
        parent, type: Human
        number, value:
            [Int literal 1234]
        money, type: int, value:
            [Int literal 120]

]

[Structure declaration Baz
    fields:
        name, type: int
        parent, type: Human
        number, value:
            [Int literal 1234]
        money, type: int, value:
            [Int literal 120]

    [Fn declaration bar, type void, arity 2
        params:
            self
            a, type int
        body:
            [Block]
                [Print]
                    [Identifier a]
    ]
    [Fn declaration buz, type Self, arity 1
        params:
            c, type bool
        body:
            [Block]
    ]
]
